woocommerce/woocommerce-ios
# Refactor of SiteCountry, that now use SelectedSiteSettings

## Description
In a [previous PR](https://github.com/woocommerce/woocommerce-ios/pull/2790), I introduced the `SelectedSiteSettings` class where I moved all the logic about fetching the latest Site Settings and takes care also to update the Currency Settings used in the app.
Since `SiteCountry` was using `ResultsController` for fetching the interested site setting, I substituted it with `SelectedSiteSettings` and I added also a test case since it's easy to introduce a regression here.

## Testing (Sanity Check)
1. Navigate to an order detail.
2. Press "Add Tracking" (you need the shipping extension on your store).
3. Fill the shipping carrier, the tracking number, and date shipped. Save it. 
4. Everything should work like before.


Update release notes:
